R/course.R

```diff
@@ -0,0 +1,29 @@
+download_zip <- function(url) {
+  dl <- curl::curl_fetch_memory(url)
+
+  httr::stop_for_status(dl$status_code)
+  stopifnot(
+    grepl("^https://dl.dropboxusercontent.com/content_link_zip/", dl$url) ||
+      grepl("^https://codeload.github.com", dl$url)
```

Comment:
I figure this should be very rigid for now.

Reply:
If you are attaching httr, why are you not using it to download the file?

Reply:
I think the answer to my question is to auto-find the filename.

I guess it feels like this code should really be in httr, not here, maybe as a extension on `httr::write_disk()`.

If we do keep it here, because we don't plan on a httr release in the near future, I think it would be better to write a `write_disk()` function and use it in a `httr::GET()` request.

Reply:
Doing this *outside* of httr, i.e. without access to unexported functions, doesn't look terribly practical. I think I can but it will definitely take me longer. How important is this @jimhester?

Reply:
Not that important, but how important is it too have this auto naming in usethis in the first place?

Reply:
Very important for this workshop use case!

I know it's hard to believe, but getting a large group of people to download a specific set of files to a folder on their computer that

a) has a sane name
b) they can find again
c) includes ALL the files and no other files

is astonishingly difficult. And I am going to get them to do it, so help me God. Or die trying.